src/main/java/br/com/caelum/viagens/administrativo/controller/PaisController.java

```diff
@@ -0,0 +1,58 @@
+package br.com.caelum.viagens.administrativo.controller;
+
+import java.net.URI;
+import java.util.Optional;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.InitBinder;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import br.com.caelum.viagens.administrativo.controller.dto.input.NewPaisInputDto;
+import br.com.caelum.viagens.administrativo.controller.dto.output.PaisOutputDto;
+import br.com.caelum.viagens.administrativo.model.Pais;
+import br.com.caelum.viagens.administrativo.repository.PaisRepository;
+import br.com.caelum.viagens.administrativo.validator.NewPaisValidator;
+
+@RestController
+@RequestMapping("/paises")
+public class PaisController {
+
+	@Autowired
+	private PaisRepository paisRepository;
+
+	@InitBinder("newPaisInputDto")
+	public void InitBinder(WebDataBinder webDataBinder) {
+		webDataBinder.addValidators(new NewPaisValidator(this.paisRepository));
+	}
+
+	@PostMapping
+	public ResponseEntity<PaisOutputDto> cadastro(@Valid @RequestBody NewPaisInputDto newPais, UriComponentsBuilder uriBuilder) {
+		Pais pais = newPais.toModel();
+		this.paisRepository.save(pais);
+
+		URI location = uriBuilder.path("/paises/{id}")
+				.buildAndExpand(pais.getId()).toUri();
+
+		return ResponseEntity.created(location).body(new PaisOutputDto(pais));
+	}
+	
+	@GetMapping("/{id}")
+	public ResponseEntity<PaisOutputDto> detalhesPais(@PathVariable("id") Optional<Pais> pais){
```

Comment:
Aqui vai ser um comentário com sentimento misto da minha parte. 

Por um lado você seguiu isso aqui *Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Usou o converter do Spring Data JPA para carregar o objeto em função do id. E realmente a gente teria que escrever esse código. 

Por outro lado, esse é um código meio mágico. Você tem um id e o parâmetro é do tipo Pais. Como eu sou bem pró usar o que tá pronto, vamos para frente. Mas tenho uma sensação que esse código é mais mágico do que precisava :).